Consolidate multiple commits into one: Add PVC manager functionality and Fix xfs_qouta clear projid

laik opened the pull request:
**Title:** Introduce PVC Manager for High-Performance LocalPV Provisioning

**Description:**

This PR introduces a new architecture for the OpenEBS Dynamic LocalPV Provisioner, addressing performance bottlenecks identified during stress testing (see https://github.com/openebs/openebs/issues/4050). The core change replaces the traditional helper pod mechanism with a DaemonSet-based PVC Manager service that performs volume operations via direct HTTP API calls.

**Key Changes:**

1.  **New PVC Manager Component:**
    *   A new [pvc-manager](file:///Users/dxp/workspace/go/src/github.com/laik/dynamic-localpv-provisioner/bin/pvc-manager/pvc-manager) binary is added (`cmd/pvc-manager/`) which runs as a DaemonSet on each node.
    *   It exposes HTTP endpoints for creating directories, applying quotas, and deleting volumes.
    *   This eliminates the overhead of creating and terminating ephemeral helper pods for each operation.

2.  **Provisioner Integration:**
    *   The main provisioner logic (`cmd/provisioner-localpv/`) is updated to conditionally use the new PVC Manager.
    *   A new client ([cmd/provisioner-localpv/app/pvc_manager_client.go](file:///Users/dxp/workspace/go/src/github.com/laik/dynamic-localpv-provisioner/cmd/provisioner-localpv/app/pvc_manager_client.go)) handles communication with the PVC Manager service.
    *   New helper functions ([cmd/provisioner-localpv/app/helper_pvc_manager.go](file:///Users/dxp/workspace/go/src/github.com/laik/dynamic-localpv-provisioner/cmd/provisioner-localpv/app/helper_pvc_manager.go)) orchestrate the HTTP calls.
    *   Environment variables `OPENEBS_IO_ENABLE_PVC_MANAGER` (default `true`) and `OPENEBS_IO_PVC_MANAGER_PORT` (default `8080`) control this behavior.

3.  **Deployment Updates:**
    *   New Kubernetes manifests are added for deploying the PVC Manager DaemonSet and its associated RBAC ([deploy/kubectl/pvc-manager-*](file:///Users/dxp/workspace/go/src/github.com/laik/dynamic-localpv-provisioner/deploy/kubectl/pvc-manager-rbac.yaml), [deploy/helm/charts/templates/pvc-manager-*](file:///Users/dxp/workspace/go/src/github.com/laik/dynamic-localpv-provisioner/deploy/helm/charts/templates/pvc-manager-service.yaml)).
    *   Documentation is provided in [design/pvc-manager-architecture.md](file:///Users/dxp/workspace/go/src/github.com/laik/dynamic-localpv-provisioner/design/pvc-manager-architecture.md) and [deploy/helm/charts/PVC-MANAGER-INTEGRATION.md](file:///Users/dxp/workspace/go/src/github.com/laik/dynamic-localpv-provisioner/deploy/helm/charts/PVC-MANAGER-INTEGRATION.md).

**Benefits:**

*   **Significantly Improved Performance:** As highlighted in issue #4050, this change drastically reduces PVC binding times, especially under load, by removing the pod creation bottleneck.
*   **Reduced API Server Load:** Fewer transient pod objects are created and managed by the Kubernetes API server.
*   **Enhanced Reliability:** A long-running service is generally more predictable than ephemeral pods.
*   **Better Resource Utilization:** The DaemonSet model provides a more consistent resource footprint.

This new architecture provides a substantial performance improvement for LocalPV provisioning, directly addressing the concerns raised in issue #4050. Backward compatibility with the helper pod mode is maintained via the `OPENEBS_IO_ENABLE_PVC_MANAGER` environment variable.

laik committed:
Consolidate multiple commits into one: Add PVC manager functionality and related fixes

Signed-off-by: laik <laik.lj@me.com>

niladrih commented:
Hi @laik. I'm yet to review your PR. Just letting you know that we require DCO signatures on all commits. You might have already noticed the failing CI job.

laik commented:
> Hi @laik. I'm yet to review your PR. Just letting you know that we require DCO signatures on all commits. You might have already noticed the failing CI job.

Yes, I saw it, it was added

laik committed:
Supplement workflows configures

Signed-off-by: laik <laik.lj@me.com>

laik committed:
Trigger CI pipeline

Empty commit to trigger CI build

Signed-off-by: laik <laik.lj@me.com>

laik committed:
Fix pvc-manager Docker build error in GitHub Actions

The pvc-manager Dockerfile expects the binary to be present in the build context, but the GitHub Actions workflow was not copying it there. This change ensures the binary is built and copied to the correct location before the Docker build step.

Signed-off-by: laik <laik.lj@me.com>

tiagolobocastro commented on examples/pvc-manager-deployment/test-pvc-manager.sh:
```diff
@@ -0,0 +1,260 @@
+#!/bin/bash
```
```suggestion
#!/use/bin/env bash
```

tiagolobocastro reviewed (commented):
Did you consider using the distributed CSI node-plugin approach rather than a custom service?
Could you please add the xfs_quota fix as a separate commit?

tiagolobocastro commented:
The image build seems to be failing, could you please take a look?

laik commented:
> The image build seems to be failing, could you please take a look?

yeah , Let me look at this issue.

> Did you consider using the distributed CSI node-plugin approach rather than a custom service? Could you please add the xfs_quota fix as a separate commit?

 What model does distributed CSI refer to?

laik committed:
try to fix github act build error

Signed-off-by: laik <laik.lj@me.com>

tiagolobocastro commented:
> What model does distributed CSI refer to?

Node deployment (--node-deployment): <https://github.com/kubernetes-csi/external-provisioner?tab=readme-ov-file#distributed-provisioning>

This is the same model used by [rawfile-localpv](https://github.com/openebs/rawfile-localpv).

laik commented:
> > What model does distributed CSI refer to?
> 
> Node deployment (--node-deployment): https://github.com/kubernetes-csi/external-provisioner?tab=readme-ov-file#distributed-provisioning
> 
> This is the same model used by [rawfile-localpv](https://github.com/openebs/rawfile-localpv).

OK, I took a look, this model will be better and can reduce the complexity of network calls, waiting for my testing and next submission. thanks a lot for the reminder
